### DISCUSSION TITLE
iOS: Create Sticker Pack via Shortcut

### DISCUSSION THREAD

**Discussion by OliverRM** [WhatsApp/stickers#802] [Q&A]
I wondered, is it possible to create a sticker pack from within a scripted shortcut using the iOS integrated shortcuts app?

Feel free to have a look at the shortcut I wrote: https://www.icloud.com/shortcuts/001c7f01c5db4aa59ee308383376835a

After WhatsApp opens, I receive the message “The operation couldn’t be completed. (com.third-party-stickers error 1000.)”. Is it generally not possible, to create sticker packs from within a shortcut or is there something wrong with my code?

**Comment by MarsVard** [comment]
There are 4 things wrong with your attempt.

1. The data type for paste bin content is not being set.
2. The json you are building is wrong.
3. Not enough stickers.
4 and you shouldn’t encode into a dictionary and then decode it again. That
step is useless. Just use the json as input for pastebin.

Kudos for the creative approach!

**Reply by OliverRM** [reply]
Thanks for your reply :)

1. How can I set it? Or is it not possible to set it from within a shortcut?
2. Oh, right. Thanks. I sent the entire root of [sticker_packs.wasticker](https://github.com/WhatsApp/stickers/blob/master/iOS/WAStickersThirdParty/sticker_packs.wasticker) only replacing `tray_image_file` with `tray_image` and `image_file` with `image_data`. It should be a single sticker pack as in [StickerPack.swift](https://github.com/WhatsApp/stickers/blob/master/iOS/WAStickersThirdParty/StickerPack.swift).sendToWhatsApp(completionHandler), right?
3. Thanks for pointing that out. I will try with more. Anyone who knows what the minimum (and maximum) is?
4. I am encoding the image data, not the dictionary.

**Comment by simonzhexu** [comment]
Using Shortcut app works. We have done that in the past.

**Reply by MarsVard** [reply]
Prove it!



j/k, I'll try to replicate it in a bit and post results
